### ISSUE TITLE
Bad conversation of "{0.attr1}{0.attr2}".format(self)

### ISSUE THREAD

**Issue by jrett2** [ikamensh/flynt#60]
(qa-new) jrett@ts-prod:~/src/qa/pylib$ flynt --verbose temp
Running flynt v.0.46
Exception 0 during conversion of code '"https://{0.hostname}{0.root}{1}" . format ( self , path )'
Traceback (most recent call last):
  File "/home/jrett/venv/qa-new/lib/python3.6/site-packages/flynt/transform/transform.py", line 29, in transform_chunk
    converted, changed, str_in_str = fstringify_node(copy.deepcopy(tree))
  File "/home/jrett/venv/qa-new/lib/python3.6/site-packages/flynt/transform/FstringifyTransformer.py", line 91, in fstringify_node
    result = ft.visit(node)
  File "/usr/lib/python3.6/ast.py", line 253, in visit
    return visitor(node)
  File "/usr/lib/python3.6/ast.py", line 308, in generic_visit
    value = self.visit(value)
  File "/usr/lib/python3.6/ast.py", line 253, in visit
    return visitor(node)
  File "/usr/lib/python3.6/ast.py", line 317, in generic_visit
    new_node = self.visit(old_value)
  File "/usr/lib/python3.6/ast.py", line 253, in visit
    return visitor(node)
  File "/home/jrett/venv/qa-new/lib/python3.6/site-packages/flynt/transform/FstringifyTransformer.py", line 30, in visit_Call
    result_node = joined_string(node)
  File "/home/jrett/venv/qa-new/lib/python3.6/site-packages/flynt/transform/format_call_transforms.py", line 92, in joined_string
    ast_name = var_map.pop(identifier)
KeyError: 0
fstringifying /home/jrett/src/qa/pylib/temp...no change

Flynt run has finished. Stats:

Execution time:                            0.016s
Files modified:                            0

_-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_._-_.
Please run your tests before committing. Did flynt get a perfect conversion? give it a star at: 
~ https://github.com/ikamensh/flynt ~
Thank you for using flynt. Upgrade more projects and recommend it to your colleagues!

(qa-new) jrett@ts-prod:~/src/qa/pylib$ cat temp
```
path='zxcv'
class self():
    hostname='asdf'
    root='qwer'

def _url(self, path):
    if path.startswith("https://"):
        return path
    else:
        return "https://{0.hostname}{0.root}{1}".format(self, path)
```

**Comment by ikamensh** [comment]
this is on purpose - this breaks factoring of taking .attr1 and .attr2 from the same variable. If it would be transformed to `f"{self.attr1}{self.attr2}"`, it would be thinkable for someone to change it later to `f"{self.attr1}{other.attr2}"` without realizing the code was designed to take attributes from the same object. Furthermore, one reason to convert to f-strings is to make code less bulky. Depending on the length of variable name, replacing single digits with multiple copies of variable name might defy this purpose and make code longer.

In short, I would suggest to use flynt to convert 95% of cases, and tackle the rest manually - you need to make some decisions on the remainder, and this can't be automated.

**Comment by ikamensh** [comment]
I see one deficiency that you can't distinguish this "failure on design" from some actual bug, I will consider how to improve here.

**Comment by jrett2** [comment]
From a user perspective, if your tool is intentionally not changing a string to an f-string (something you don't document), you should note that and move on. There is no valid time for a tool like this to raise a KeyError.
